addons/point_of_sale/models/pos_session.py

```diff
@@ -562,9 +551,10 @@ def get_closing_control_data(self):
             'opening_notes': self.opening_notes,
             'default_cash_details': {
                 'name': default_cash_payment_method_id.name,
-                'amount': self.cash_register_id.balance_start + total_default_cash_payment_amount +
-                                             sum(self.cash_register_id.line_ids.mapped('amount')),
-                'opening': self.cash_register_id.balance_start,
+                'amount': self.cash_register_balance_start
+                          + total_default_cash_payment_amount
+                          + self.cash_real_transaction,
```

Comment:
```suggestion
                          + sum(self.statement_line_ids.mapped('amount')),
```

Reply:
Note the `cash_real_transaction` is not yet assigned at this point. The enclosing method is called prior to calling `_validate_session`.